yt/frontends/ramses/hilbert.py

```diff
@@ -48,63 +49,22 @@
 
 
 def hilbert3d(
-    X: "np.ndarray[Any, np.dtype[np.float64]]", bit_length: int
+    ijk: "np.ndarray[Any, np.dtype[np.int64]]", bit_length: int
 ) -> "np.ndarray[Any, np.dtype[np.float64]]":
     """Compute the order using Hilbert indexing.
 
     Arguments
     ---------
-    X : (N, ndim) float array
+    ijk : (N, ndim) integer array
       The positions
     bit_length : integer
       The bit_length for the indexing.
     """
-    X = np.atleast_2d(X)
-
-    x_bit_mask, y_bit_mask, z_bit_mask = (
-        np.zeros(bit_length, dtype=bool) for _ in range(3)
-    )
-    i_bit_mask = np.zeros(3 * bit_length, dtype=bool)
-
-    npoint = X.shape[0]
-    order = np.zeros(npoint)
-
-    # Convert positions to binary
-    for ip in range(npoint):
-        for i in range(bit_length):
-            mask = 0b01 << i
-            x_bit_mask[i] = X[ip, 0] & mask
-            y_bit_mask[i] = X[ip, 1] & mask
-            z_bit_mask[i] = X[ip, 2] & mask
-
-        for i in range(bit_length):
-            # Interleave bits
-            i_bit_mask[3 * i + 2] = x_bit_mask[i]
-            i_bit_mask[3 * i + 1] = y_bit_mask[i]
-            i_bit_mask[3 * i] = z_bit_mask[i]
-
-        # Build Hilbert ordering using state diagram
-        cstate = 0
-        for i in range(bit_length - 1, -1, -1):
-            sdigit = (
-                4 * i_bit_mask[3 * i + 2]
-                + 2 * i_bit_mask[3 * i + 1]
-                + 1 * i_bit_mask[3 * i]
-            )
-            nstate = _STATE_DIAGRAM[sdigit, 0, cstate]
-            hdigit = _STATE_DIAGRAM[sdigit, 1, cstate]
-
-            i_bit_mask[3 * i + 2] = hdigit & 0b100
-            i_bit_mask[3 * i + 1] = hdigit & 0b010
-            i_bit_mask[3 * i] = hdigit & 0b001
-
-            cstate = nstate
-
-        # Compute ordering
-        for i in range(3 * bit_length):
-            order[ip] = order[ip] + i_bit_mask[i] * 2**i
-
-    return order
+    ijk = np.atleast_2d(ijk)
+    # A note here: there is a freedom in the way hilbert indices are
+    # being computed (should it be xyz or yzx or zxy etc.)
```

Comment:
Is the yt convention just the reverse?  (i.e., `xyz` to `zyx`) or are we otherwise shuffling?

Reply:
I _think_ yt's convention is either xyz (or zyx) and RAMSES' seems to be xzy (or yzx). In any ways, it's just a permutation that needs to be done once and for all.